### PR TITLE
 Add at_front option for decorated functions

### PR DESCRIPTION
**Pull request by Pavel-Guseynov**

### CODE REVIEW

File: rq/queue.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def fetch_job(self, job_id):
         except NoSuchJobError:
             self.remove(job_id)
         else:
-            if job.origin == self.name:
+            if job.origin == self.name or 'failed':
```
Comment (selwin) [inline rq/queue.py]: What's the purpose of this change?

**Comment by theodesp** [comment]
@Pavel-Guseynov Can you elaborate more on the reasoning behind this PR?

**Comment by selwin** [comment]
Superseded by https://github.com/nvie/rq/pull/888
